base/abstractarraymath.jl

```diff
@@ -200,31 +200,61 @@ function repmat(a::AbstractVector, m::Int)
     return b
 end
 
-# Generalized repmat
-function repeat{T}(A::AbstractArray{T};
-                   inner::Array{Int} = ones(Int, ndims(A)),
-                   outer::Array{Int} = ones(Int, ndims(A)))
+"""
+    repeat(A::AbstractArray; inner=(), outer=())
```

Comment:
Not sure which implementation should be preferred. In terms of making the default in the docs match the more complicated version, how about giving the default value some temporary variable name that gets explained in more detail in the docstring body?